### DISCUSSION TITLE
Trying to change password of a device without expecting prompt

### DISCUSSION THREAD

**Discussion by reddyarun474** [ktbyers/netmiko#2449] [Q&A]
There is a usecase in my environment, by default device will have some default credentials and on login it will ask to change default credentials. Below is the manual snippet for the same.

Manual steps:

Using username "root".
Keyboard-interactive authentication prompts from server:
End of keyboard-interactive prompts from server
Access denied
Keyboard-interactive authentication prompts from server:
| Password:
| You are required to change your password immediately (root enforced)
| Changing password for root.
| (current) UNIX password:
| Enter new UNIX password:
| Retype new UNIX password:

I am trying to automate above manual steps using netmiko and protocol is ssh. Can anyone give some pointers here ?

**Comment by ktbyers** [comment]
The pattern shown here should probably work:

https://github.com/ktbyers/netmiko/blob/develop/EXAMPLES.md#handling-commands-that-prompt-timing

If you are using Netmiko4 (currently the develop branch), you could also use send_multiline_timing()

**Comment by reddyarun474** [comment]
Tried as suggested, but it is not working as expected. Won't we expect any prompt after login, if yes, then we will see prompt as "(current) UNIX password:"

BUG:paramiko.transport:starting thread (client mode): 0xb16c7dc0
DEBUG:paramiko.transport:Local version/idstring: SSH-2.0-paramiko_2.7.2
DEBUG:paramiko.transport:Remote version/idstring: SSH-2.0-OpenSSH_7.4
INFO:paramiko.transport:Connected (version 2.0, client OpenSSH_7.4)
DEBUG:paramiko.transport:kex algos:['ecdh-sha2-nistp256', 'ecdh-sha2-nistp384', 'diffie-hellman-group14-sha1', 'diffie-hellman-group-exchange-sha1'] server key:['ssh-rsa', 'rsa-sha2-512', 'rsa-sha2-256', 'ecdsa-sha2-nistp256', 'ssh-ed25519'] client encrypt:['aes128-gcm@openssh.com', 'aes256-gcm@openssh.com', 'aes128-cbc', 'aes256-cbc', 'aes128-ctr', 'aes192-ctr', 'aes256-ctr', 'aes192-cbc'] server encrypt:['aes128-gcm@openssh.com', 'aes256-gcm@openssh.com', 'aes128-cbc', 'aes256-cbc', 'aes128-ctr', 'aes192-ctr', 'aes256-ctr', 'aes192-cbc'] client mac:['hmac-sha1'] server mac:['hmac-sha1'] client compress:['none', 'zlib@openssh.com'] server compress:['none', 'zlib@openssh.com'] client lang:[''] server lang:[''] kex follows?False
DEBUG:paramiko.transport:Kex agreed: ecdh-sha2-nistp256
DEBUG:paramiko.transport:HostKey agreed: ssh-ed25519
DEBUG:paramiko.transport:Cipher agreed: aes128-ctr
DEBUG:paramiko.transport:MAC agreed: hmac-sha1
DEBUG:paramiko.transport:Compression agreed: none
DEBUG:paramiko.transport:kex engine KexNistp256 specified hash_algo <built-in function openssl_sha256>
DEBUG:paramiko.transport:Switch to new keys ...
DEBUG:paramiko.transport:Adding ssh-ed25519 host key for x.x.x.x: b'57a3a227b850bdedff5fd70721a67d00'
DEBUG:paramiko.transport:userauth is OK
DEBUG:paramiko.transport:Authentication type (password) not permitted.
DEBUG:paramiko.transport:Allowed methods: ['publickey', 'keyboard-interactive', 'hostbased']
DEBUG:paramiko.transport:userauth is OK
INFO:paramiko.transport:Authentication (keyboard-interactive) failed.
DEBUG:paramiko.transport:EOF in transport thread

**Reply by ktbyers** [reply]
It looks like your device is failing authentication.
